lib/tasks/obo.rake

```diff
@@ -19,5 +20,116 @@ namespace :obo do
+      oleracea: [:acephala, 'acephala?', :viridis, :africana, :capitata, :capitata?,
+                 :gemnifera, :sabauda, :sabellica, :tronchuda, :tronchuda?, :alboglabra?],
+      rapa: ['Broccoletto Gp', 'Neep Greens Gp', 'Pak Choi Group', :chinensis, :dichotoma,
+             :oleifera, :pekinensis, :purpurea, :ruvo, :sylvestris, :trilocularis],
+      rapa?: [:rapa?],
+      repanda: [:almeriensis, :blancoana, :cadevallii, :cantabrica, :confusa, :maritima, :nudicaulis],
+      rupestris: [:glaucescens, :hispida],
+      souliei: [:dimorpha],
+      spinescens: [:spinescens],
+      villosa: [:tinei, :villosa]
+    }
+    varieties_count = 0
+    varieties_input.each do |species, varieties|
+      species_term = TaxonomyTerm.find_by(name: "Brassica #{species}")
+      varieties.each do |variety|
+        tax_term = find_variety('Brassica', species, variety)
+        unless tax_term
+          # YES, we don't use var. or subsp., since checking those will waste too much time...
+          tax_term = TaxonomyTerm.
+                     new(name: "Brassica #{species} #{variety}")
+        end
+        save_with_parent(tax_term, species_term)
+        varieties_count += 1
+      end if species_term
+    end
+
+    puts "  Added #{varieties_count} varieties."
+
+
+    cultivars_input = {
+      juncea: {
+        integrifolia: [:crispifolia, :integrifolia, :rugosa]
+      },
+      napus: {
+        napus: [:annua, :biennis, :pabularia],
+        oleifera: [:annua, :biennis]
+      },
+      oleracea: {
+        acephala: [:medullosa, :palmifolia, :ramosa, :sabellica, :viridis],
+        botrytis: [:botrytis, :italica],
+        capitata: [:rubra, :sabauda, :alba]
+      },
+      rapa: {
+        chinensis: [:communis, :parachinensis, :purpuraria, :rosularis],
+        oleifera: [:annua, :biennis, :sylvestris],
+        pekinensis: [:glabra, :laxa, :pekinensis, :pandurata]
+      }
+    }
+    cultivars_count = 0
+    cultivars_input.each do |species, varieties|
+      varieties.each do |variety, cultivars|
+        variety_term = TaxonomyTerm.
+                       find_by(name: variety_names('Brassica', species, variety))
+        cultivars.each do |cultivar|
+          tax_term = TaxonomyTerm.
+                     find_or_initialize_by(name: "#{variety_term.name} #{cultivar}")
+          save_with_parent(tax_term, variety_term)
+          cultivars_count += 1
+        end if variety_term
+      end
+    end
+
+    puts "  Added #{cultivars_count} cultivars"
+  end
+
+
+  def variety_names(genus, species, subtaxa)
+    ["#{genus} #{species} var. #{subtaxa}",
+     "#{genus} #{species} subsp. #{subtaxa}",
+     "#{genus} #{species} #{subtaxa}"]
+  end
+
+  def find_variety(genus, species, subtaxa)
+    TaxonomyTerm.
+      where(name: variety_names(genus, species, subtaxa)).
+      first
+  end
+
+  def save_with_parent(term, parent)
+    term.canonical = false
+    term.label = 'CROPSTORE'
+    term.parent = parent
+    term.save
```

Comment:
Minor thingie, most likely not important in this case - I like to use bang version by default when saving records (`save!`, `create!`, `update_attributes!`) unless I am checking return value or validity.
Do not mind me if I mentioned that already, I do that often ;)